tikv-client/src/main/java/com/pingcap/tikv/key/KeyRangeBuilder.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2019 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pingcap.tikv.key;
+
+import com.google.common.collect.BoundType;
+import com.google.common.collect.Range;
+import com.pingcap.tikv.predicates.IndexRange;
+
+class KeyRangeBuilder {
```

Comment:
This class can be an abstract class with `abstract Pair<Key, Key> compute()` method. The benefit is we cannot directly use `KeyRangeBuilder` but use a concrete one.

Reply:
we cannot use `abstract Pair<Key, Key> compute()` because compute() returns different types in different classes extended

Reply:
I can make KeyRangeBuilder abstract though.

Reply:
that works too.

Reply:
done.